lib/adapters/mongodb/index.js

```diff
@@ -89,6 +89,29 @@ module.exports = function (mongodbUrl, options) {
                 }
                 return findById(type, id)
             })
+        }
+
+    const extendPatch = function(patch, data, parent) {
+        Object.keys(data).forEach(function(key){
+            if (typeof data[key] === 'object') {
+                patch = extendPatch(patch, data[key], parent + key + '.');
+            } else {
+                patch[parent + key] = data[key];
+            }
+        })
+        return patch;
+    }
+
+    const patch = function(type, id, data) {
+        const model = models[type]
+        const patchObject = extendPatch({}, data, '');
```

Comment:
Perhaps you could add some comments to this, like you did in the PR description. It's a bit non obvious what the purpose is.  

I see you created your own function to convert from object to a dot syntax, maybe we can leverage an existing lib to do that for us? Here is one which I found, but others exist I'm sure : https://www.npmjs.com/package/dot-object#convert-object-to-dotted-keyvalue-pair